### PR TITLE
docs: add cli user guide

### PR DESCRIPTION
**Pull request by Yihen-Liu**
Signed-off-by: liuyihen@onchain.com

### CODE REVIEW

File: docs/specifications/cli_user_guide.md [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+hex address:    	 01cde971fcedd97fba7babd19516a6d90db1ed97
+base58 address: 	 TA8kMG3QZu37NmPNCPuP1xNPYbmiFTX973
+
+NOTE: 
+If without wallet name, the default value wallet.dat will be used.
+```
+
+### Example for wallet balance
+```
+$ ./ontology wallet show --name wallet.dat
```
Comment (qiluge) [inline docs/specifications/cli_user_guide.md]: 这个命令是wallet show，不是显示钱包余额的命令

File: docs/specifications/cli_user_guide.md [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+        --cname      value               contract name that will be deployed
+        --cversion   value               contract version which will be deployed
+        --author     value               owner of deployed smart contract
+        --email      value               owner email who deploy the smart contract
+        --desc       value               contract description when deploy one
+```
+
+### Example for contract deploy
+```
+$ ./contract deploy --type=1 --store=true --code=./deploy.txt --cname=testContract --cversion=0.0.1 --author=Yihen --desc="this is my first test contract" --email="name@emailaddr.com"
```
Comment (qiluge) [inline docs/specifications/cli_user_guide.md]: ./ontology contract，不是./contract

**Review by qiluge — COMMENTED** [review]
有两条命令写错了

File: docs/specifications/cli_user_guide.md (hunk 1 above)
Comment (qiluge) [reply docs/specifications/cli_user_guide.md]: ./ontology wallet balance --name wallet.dat

Comment (Yihen-Liu) [reply docs/specifications/cli_user_guide.md]: Thx， have fix.

**Review by laizy — APPROVED** [review]

File: docs/specifications/cli_user_guide.md [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+Description:
+    With this command, you can control assert through transaction.
+
+Command:
+    transfer
+        --caddr     value                 smart contract address
+        --from      value                 wallet address base58, which will transfer from
+        --to        value                 wallet address base58, which will transfer to
+        --value     value                 how much asset will be transfered
+        --password  value                 use password who transfer from
```
Comment (javajoker) [inline docs/specifications/cli_user_guide.md:289]: use plaintext password is not safe in most cases. better input password without showing them.

anyway, if the enduser insists on using plaintext, e.g., in some shell script, this --password parameter can be applied.

please add more comment on this parameter.

Comment (Yihen-Liu) [reply docs/specifications/cli_user_guide.md:289]: Hi Ning, Thanks for your suggestion. 
I have change the implement that, user can choose to input password after program is run.

Comment (laizy) [reply docs/specifications/cli_user_guide.md:289]: relevant pr #52

**Pull request merged**